### ISSUE TITLE
Error restoring primary key: relation already exists

### ISSUE THREAD

**Issue by lospejos** [dimitri/pgcopydb#220]
Error `relation already exists` appears when restoring primary key for a table.

Interesting that SQL logged in error, does not contain `using index xxx` clause (other primary key creation SQL statements have `using index xxx` clause).

Multiple attempts leads to this error on the same table each time, so most probably this is not a multi-threading/thread overlapping issue.

Also we tried to run with `--index-jobs 1` option, the same error still present.

As we see from https://pgcopydb.readthedocs.io/en/latest/design.html#for-each-table-build-all-indexes-concurrently `pgcopydb` uses advanced approach to create indexes: first all primary keys created as unique indexes, then second statement creates primary key based on unique index.

The unique index on which primary key should be created is present in target database.

Also for some other tables in the same schema, primary key restoration is performed successfully.
In logs we see unique index creation (named as primary index, `_pkey` suffix), then

`alter table xxx add primary key xxx_pkey using index xxx_pkey`

successfully executed.

Please assist.

Thanks.

**Comment by lospejos** [comment]
From 
https://github.com/dimitri/pgcopydb/blob/5abfc833253e7019f2d86b9a694b437bd78a87fe/src/bin/pgcopydb/indexes.c#L985

we see, that if index to be restored is primary or unique, the `USING INDEX`  clause will be added to index creation query.

In our target database the created index (on which primary key should be created) is unique.

**Comment by lospejos** [comment]
Another notice:

When indexes was successfully created using unique key, SQL statement was like
```
ALTER TABLE "sss"."ttt" ADD CONSTRAINT "ttt_pkey" PRIMARY KEY USING INDEX "ttt_pkey";
```

But when the occurred, SQL statement contained `ONLY` keyword:

```
ALTER TABLE ONLY sss.ttt1 ADD CONSTRAINT ttt1_pkey PRIMARY KEY (ttt1_id);
```

**Comment by dimitri** [comment]
Hi @lospejos ; if you want help you need to share exact details about the problem you are seeing, in a way that I can understand, evaluate and reproduce the problem. Your description is a good start, but there is nothing I can do with it. Can you please share the logs of the operation you're running, and maybe even a test schema that I could use where the problem happens?

**Comment by lospejos** [comment]
> Hi @lospejos ; if you want help you need to share exact details about the problem you are seeing, in a way that I can understand, evaluate and reproduce the problem. Your description is a good start, but there is nothing I can do with it. Can you please share the logs of the operation you're running, and maybe even a test schema that I could use where the problem happens?

Thanks for your quick response!

After some research it seems that I've found a cause: if our table (or related table) contains columns constraints using `btree_gist` extension features, this (or similar) error appears during data migration attempt:

```
pg_restore: error: could not execute query: ERROR:  relation "tbl_1_ext_id_key" already exists
Command was: ALTER TABLE ONLY loy1.tbl_1 ADD CONSTRAINT tbl_1_ext_id_key UNIQUE (ext_id);
```

This is not the same error as it was mentioned above (the earlier one complains about `primary key` creation, the latter one complains about `unique` constraint creation), but it seems that both of these error have the same or similar cause.

Both of these errors appears after executing SQL statement containing `ALTER TABLE ONLY...` which is not produced by `pgcopydb` (at least, I didn't found any code generating this `ONLY` clause in `ALTER TABLE` statement).

Anyway, this is just my assumption and the real error cause may be different.

I prepared test schema: 
[schema_ddl.sql.txt](https://github.com/dimitri/pgcopydb/files/11091986/schema_ddl.sql.txt)


which is attached to this comment (file has `txt` extension since github doesn't accept files with `sql` extension).

If you try to migrate this schema, I suppose you will see this error.

Thanks!
